jgrapht-core/src/main/java/org/jgrapht/alg/cycle/AhujaOrlinSharmaCyclicExchangeLocalAugmentation.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * (C) Copyright 2003-2018, by Christoph Grüne and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+
+import java.util.*;
+
+/**
+ * Implementation of an algorithm for the local augmentation problem for the cyclic exchange neighborhood,
+ * i.e. it finds subset-disjoint negative cycles, based on the
+ * <a href ="http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.15.6758&rep=rep1&type=pdf">paper</a>
+ * by Ahuja et al.
+ *
+ * This algorithm may enumerate all paths up to the length given by the parameter <code>lengthBound</code>,
+ * i.e the algorithm runs in exponential time.
+ *
+ * @param <V> the vertex type
+ * @param <E> the edge type
+ *
+ * @author Christoph Grüne
+ * @since June 7, 2018
+ */
+public class AhujaOrlinSharmaCyclicExchangeLocalAugmentation<V, E> {
+
+    private Graph<V, E> graph;
+    private Map<V, Integer> labels;
+    private int lengthBound;
+
+    /**
+     * Constructs an algorithm with given inputs
+     *
+     * @param graph the (improvement) graph on which to calculate the local augmentation
+     * @param lengthBound the upper bound for the length of cycles to detect
+     * @param labels the labels of the vertices encoding the subsets of vertices
+     */
+    public AhujaOrlinSharmaCyclicExchangeLocalAugmentation(Graph<V, E> graph, int lengthBound, Map<V, Integer> labels) {
+        this.graph = graph;
+        this.lengthBound = lengthBound;
+        this.labels = labels;
+    }
+
+    /**
+     * calculates a valid subset-disjoint negative cycle.
+     * If there is no such a cycle, it returns an empty LabeledPath instance with cost Double.MAX_VALUE.
+     *
+     * @return a valid subset-disjoint negative cycle encoded as LabeledPath
+     */
+    public LabeledPath<V> getLocalAugmentationCycle() {
+
+        int k = 1;
+        LabeledPath<V> C = new LabeledPath<>();
```

Comment:
Variable names should always start with a lowercase letter; only class names start with an uppercase letter.  (This applies to some of your other pull requests as well.)

Reply:
This is true and I know the convention but I want to use the notation of the paper. However, I will change this here.